amazonia/classes/asg.py

```diff
@@ -145,6 +151,19 @@ def create_launch_config(self, title, keypair, image_id, instance_type, iam_inst
         if iam_instance_profile_arn is not None:
             self.lc.IamInstanceProfile = iam_instance_profile_arn
         self.lc.UserData = Base64(userdata)
+
+        # TODO: Work out how to avoid the assumption here of an amazon linux OS.
+        if hdd_size is not None:
+            self.lc.BlockDeviceMappings = [
+                ec2.BlockDeviceMapping(
+                    DeviceName="/dev/xvda",
+                    Ebs=ec2.EBSBlockDevice(
+                        VolumeSize=hdd_size,
+                        VolumeType='gp2'
+                    )
+                ),
+            ]
+
         return launch_config_title
 
     def create_cd_deploygroup(self, title, cd_service_role_arn):
```

Comment:
Also update this one to http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codedeploy-application.html
and add http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codedeploy-deploymentgroup.html